omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/SessionController1_9.java

```diff
@@ -64,27 +67,42 @@ public Object get(WebRequest request) {
 			session.add("user", ConversionUtil.convertToRepresentation(Context.getAuthenticatedUser(), rep));
 			session.add("locale", Context.getLocale());
 			session.add("allowedLocales", Context.getAdministrationService().getAllowedLocales());
+			session.add("sessionLocation",
+			    ConversionUtil.convertToRepresentation(Context.getUserContext().getLocation(), rep));
 		}
 		return session;
 	}
 	
 	@RequestMapping(method = RequestMethod.POST)
 	@ResponseBody
 	@ResponseStatus(value = HttpStatus.OK)
-	public void post(@RequestBody Map<String, String> body) {
+	public void post(HttpServletRequest request, @RequestBody Map<String, String> body) {
 		String localeStr = body.get("locale");
-		Locale locale = null;
-		try {
-			locale = LocaleUtils.toLocale(localeStr);
-		}
-		catch (IllegalArgumentException e) {
-			throw new APIException(" '" + localeStr + "' does not represent a valid locale.");
+		if (localeStr != null) {
+			Locale locale = null;
+			try {
+				locale = LocaleUtils.toLocale(localeStr);
+			}
+			catch (IllegalArgumentException e) {
+				throw new APIException(" '" + localeStr + "' does not represent a valid locale.");
+			}
+			Set<Locale> allowedLocales = new HashSet<Locale>(Context.getAdministrationService().getAllowedLocales());
+			if (allowedLocales.contains(locale)) {
+				Context.setLocale(locale);
+			} else {
+				throw new APIException(" '" + localeStr + "' is not in the list of allowed locales.");
+			}
 		}
-		Set<Locale> allowedLocales = new HashSet<Locale>(Context.getAdministrationService().getAllowedLocales());
-		if (allowedLocales.contains(locale)) {
-			Context.setLocale(locale);
-		} else {
-			throw new APIException(" '" + localeStr + "' is not in the list of allowed locales.");
+		String locationUuid = body.get("sessionLocation");
+		if (locationUuid != null) {
```

Comment:
Assuming there are use cases where sessionLocation is optional, do we want to provide a mechanism to "unset" a previously selected sessionLocation?  I'm not 100% sure the proper way to handle this, but it would seem reasonable to check if "sessionLocation" is found as an attribute, but a null or empty value is set for it, then set the sessionLocation on the session to null.   Otherwise, I think this PR looks good to merge.

Reply:
So the RESTful way to do it would be to have routes GET/POST/DELETE `/session/locale`, `session/location`, etc. But I think it would be fine just to let `"sessionLocation":""` mean "unset."

Reply:
Ah, OpenMRS Core UserContext doesn't provide a way to unset location. See https://github.com/openmrs/openmrs-core/blob/6082fd6dc6c3c0c2b3ea78aab7143f0dff855215/api/src/main/java/org/openmrs/api/context/UserContext.java#L422

Reply:
Fair enough.  Seems like something we might want to fix in core, but out of scope for this ticket...